src/test/java/org/fogbowcloud/manager/core/processors/SpawningProcessorTest.java

```diff
@@ -60,243 +55,279 @@ public void setUp() {
 
-        LocalCloudConnector localCloudConnector = mock(LocalCloudConnector.class);
+        PowerMockito.mockStatic(CloudConnectorFactory.class);
+        BDDMockito.given(CloudConnectorFactory.getInstance()).willReturn(cloudConnectorFactory);
 
-        CloudConnectorFactory cloudConnectorFactory = mock(CloudConnectorFactory.class);
-        when(cloudConnectorFactory.getCloudConnector(anyString())).thenReturn(localCloudConnector);
+        LocalCloudConnector localCloudConnector = Mockito.mock(LocalCloudConnector.class);
+        Mockito.when(cloudConnectorFactory.getCloudConnector(Mockito.anyString()))
+                .thenReturn(localCloudConnector);
 
-        PowerMockito.mockStatic(CloudConnectorFactory.class);
-        given(CloudConnectorFactory.getInstance()).willReturn(cloudConnectorFactory);
+        this.cloudConnector = CloudConnectorFactory.getInstance()
+                .getCloudConnector(BaseUnitTests.LOCAL_MEMBER_ID);
 
-        this.cloudConnector = CloudConnectorFactory.getInstance().getCloudConnector(
-                BaseUnitTests.LOCAL_MEMBER_ID);
+        this.tunnelingService = Mockito.mock(TunnelingServiceUtil.class);
+        this.sshConnectivity = Mockito.mock(SshConnectivityUtil.class);
 
-        this.spawningProcessor = spy(new SpawningProcessor(
-                BaseUnitTests.LOCAL_MEMBER_ID, this.tunnelingService, this.sshConnectivity,
+        this.spawningProcessor = Mockito.spy(new SpawningProcessor(BaseUnitTests.LOCAL_MEMBER_ID,
+                this.tunnelingService, this.sshConnectivity,
                 DefaultConfigurationConstants.SPAWNING_ORDERS_SLEEP_TIME));
 
         SharedOrderHolders sharedOrderHolders = SharedOrderHolders.getInstance();
         this.spawningOrderList = sharedOrderHolders.getSpawningOrdersList();
         this.fulfilledOrderList = sharedOrderHolders.getFulfilledOrdersList();
         this.failedOrderList = sharedOrderHolders.getFailedOrdersList();
         this.openOrderList = sharedOrderHolders.getOpenOrdersList();
-        
+
         this.thread = null;
     }
-    
+
     @After
     public void tearDown() {
         if (this.thread != null) {
             this.thread.interrupt();
         }
         super.tearDown();
     }
-    
+
+    // test case: In calling the processSpawningOrder() method for any order other than spawning,
+    // you must not make state transition by keeping the order in your source list.
     @Test
-    public void testTryingProcessComputeOrderNotSpawning() throws Exception {
+    public void testProcessComputeOrderNotSpawning() throws Exception {
+
+        // set up
         Order order = spyComputeOrder();
         order.setOrderState(OrderState.OPEN);
         this.openOrderList.addItem(order);
-        
+
+        // exercise
         this.spawningProcessor.processSpawningOrder(order);
 
-        assertNotNull(this.openOrderList.getNext());
-        assertNull(this.fulfilledOrderList.getNext());
+        // verify
```

Comment:
@wassoares I think it is better to check the order in the list is the one we expect instead of checking if there is one order